instrumentation/jaxrs2/src/test/java/brave/jaxrs2/ITTracingFeature_Container.java

```diff
@@ -102,18 +112,72 @@ public Response toResponse(Exception e) {
     }
   }
 
-  @Override public void init(ServletContextHandler handler) {
+  @Retention(RetentionPolicy.RUNTIME) public @interface SkipTracing {
+  }
+
+  @Path("declarative")
+  public static class Declarative { // public for resteasy to inject
 
+    @GET
+    @Path("foo")
+    @SkipTracing // based on declaration vs http path
+    public Response foo() {
+      return Response.status(200).build();
+    }
+
+    @GET
+    @Path("bar")
+    public Response bar() {
+      return Response.status(200).build();
+    }
+  }
 
+  @Test
+  public void declarative_sampling() throws Exception {
+    httpTracing = ContainerSampler.decorate(httpTracing, new RateForMethod<ResourceInfo>() {
```

Comment:
ex. here's an example of weaving in a custom trace annotation into JAX-RS cc @reta @jplock